src/primitives.es6

```diff
@@ -0,0 +1,339 @@
+/*
+ * A refract element implementation with an extensible type registry.
+ */
+
+import _ from 'lodash';
+import {convertToType, convertFromRefract, convertFromCompactRefract} from './convert';
+
+/*
+ * A private symbol for subclasses to set key names of attributes which should
+ * be converted from refract elements rather than simple types.
+ */
+export const attributeElementKeys = Symbol('attributeElementKeys');
+
+/*
+ * ElementType is the base element from which all other elements are built.
+ * It has no specific information about how to handle the content, but is
+ * able to convert to and from Refract/Javascript.
+ */
+export class ElementType {
+  constructor(element, content=null, meta={}, attributes={}) {
+    this.element = element;
+    this.meta = meta;
+    this.attributes = attributes;
+    this.content = content;
+
+    this[attributeElementKeys] = [];
+  }
+
+  toValue() {
+    return this.content;
+  }
+
+  toRefract(options={}) {
```

Comment:
I think it would be good to start following this:
https://github.com/airbnb/javascript

If you want to try to do it in this PR, feel free, but we can also merge this and try to do it in another PR.

For example, this line should be:

```javascript
toRefract(options = {}) {
```